src/coreclr/vm/perfmap.cpp

```diff
@@ -86,29 +94,121 @@ void PerfMap::Initialize()
         
         s_enabled = true;
     }
+
+    if (sendExisting)
+    {
+        if (type == PerfMapType::ALL || type == PerfMapType::JITDUMP)
+        {
+            AppDomain::AssemblyIterator assemblyIterator = GetAppDomain()->IterateAssembliesEx(
+                (AssemblyIterationFlags)(kIncludeLoaded | kIncludeExecution));
+            CollectibleAssemblyHolder<DomainAssembly *> pDomainAssembly;
+            while (assemblyIterator.Next(pDomainAssembly.This()))
+            {
+                CollectibleAssemblyHolder<Assembly *> pAssembly = pDomainAssembly->GetAssembly();
+                PerfMap::LogImageLoad(pAssembly->GetPEAssembly());
+
+                Module *pModule = pAssembly->GetModule();
+                if (pModule->IsReadyToRun())
+                {
+                    ReadyToRunInfo::MethodIterator mi(pModule->GetReadyToRunInfo());
+
+                    while (mi.Next())
+                    {
+                        // Call GetMethodDesc_NoRestore instead of GetMethodDesc to avoid restoring methods.
+                        MethodDesc *hotDesc = (MethodDesc *)mi.GetMethodDesc_NoRestore();
+                        if (hotDesc != nullptr && hotDesc->GetNativeCode() != NULL)
+                        {
+                            PerfMap::LogPreCompiledMethod(hotDesc, hotDesc->GetNativeCode());
+                        }
+                    }
+                }
+            }
+        }
+
+        {
+            CodeVersionManager::LockHolder codeVersioningLockHolder;
+
+            EEJitManager::CodeHeapIterator heapIterator(nullptr);
+            while (heapIterator.Next())
+            {
+                MethodDesc * pMethod = heapIterator.GetMethod();
+                if (pMethod == nullptr)
+                {
+                    continue;
+                }
+
+                PCODE codeStart = PINSTRToPCODE(heapIterator.GetMethodCode());
+                NativeCodeVersion nativeCodeVersion;
+#ifdef FEATURE_CODE_VERSIONING
+                nativeCodeVersion = pMethod->GetCodeVersionManager()->GetNativeCodeVersion(pMethod, codeStart);;
+                if (nativeCodeVersion.IsNull() && codeStart != pMethod->GetNativeCode())
+                {
+                    continue;
+                }
+#else // FEATURE_CODE_VERSIONING
+                if (codeStart != pMethod->GetNativeCode())
+                {
+                    continue;
+                }
+#endif // FEATURE_CODE_VERSIONING
+
+                EECodeInfo codeInfo(codeStart);
+                IJitManager::MethodRegionInfo methodRegionInfo;
+                codeInfo.GetMethodRegionInfo(&methodRegionInfo);
+                _ASSERTE(methodRegionInfo.hotStartAddress == codeStart);
+                    
+                PrepareCodeConfig config(!nativeCodeVersion.IsNull() ? nativeCodeVersion : NativeCodeVersion(pMethod), FALSE, FALSE);
+                PerfMap::LogJITCompiledMethod(pMethod, codeStart, methodRegionInfo.hotSize, &config);
```

Comment:
I agree with you that it seems wrong but it isn't an instant fix, since it already exists in rundown I filed an issue and we can fix it later: https://github.com/dotnet/runtime/issues/86086